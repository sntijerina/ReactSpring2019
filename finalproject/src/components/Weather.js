import React, { Component } from 'react';
import Grid from "@material-ui/core/Grid";
import WeatherForm from './WeatherForm';
import WeatherCard from './WeatherCard';
import axios from 'axios';

const weatherMapApiKey = '75ca74d04777316985b8b57142df457b';
const weatherMapApiUrl = `http://api.openweathermap.org/data/2.5/weather?appid=${weatherMapApiKey}&units=imperial`;
const defaultLocation = 'chicago';

class Weather extends Component {
    
    constructor(props) 
    {
      super(props);

      this.state  = { 
        name: '',
        temp: '', 
        minTemp: '', 
        maxTemp: '',
        pressure: '',
        humidity: '',
        wind: '',
        overcast: '',
        icon:'',
        error: ''
      };

      this.getWeather = this.getWeather.bind(this);
    }

    componentDidMount = ()=> { this.getWeather(defaultLocation) };

    getWeather(location) {
        let self = this;
        const encodedLocation = encodeURIComponent(location);
        let apiUrl = `${weatherMapApiUrl}&q=${encodedLocation}`;
     
        axios.get(apiUrl)
        .then(response => { 
            console.log(response);
            self.setState({ 
                name: response.data.name,
                temp: response.data.main.temp,
                minTemp: response.data.main.temp_min, 
                maxTemp: response.data.main.temp_max,
                pressure: response.data.main.pressure,
                humidity: response.data.main.humidity,
                wind: response.data.wind,
                overcast: response.data.weather[0].description,
                icon: `http://openweathermap.org/img/w/${response.data.weather[0].icon}.png`
            });
        })
        .catch(error => {
            console.log(error);
        });
    }

    render() {
      return (
        <Grid
          spacing={0}
          container
          alignItems="center"
          direction="column"
          style={{ minHeight: "100vh" }}
          justify="center"
        >
        <Grid item xs={3}>
          <WeatherForm onFormSubmit={this.getWeather}/>
          <WeatherCard
            name = { this.state.name }
            temp = { this.state.temp } 
            maxTemp = { this.state.maxTemp }
            minTemp = { this.state.minTemp }
            pressure = { this.state.pressure }
            humidity = { this.state.humidity }
            overcast = { this.state.overcast }
            icon = { this.state.icon }
            />     
        </Grid>
      </Grid>       
      );
    }
  }
  
  export default Weather;