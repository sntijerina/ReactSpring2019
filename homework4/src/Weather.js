import React, { Component } from 'react';
import axios from 'axios';

const apiKey = '75ca74d04777316985b8b57142df457b';
const url = `http://api.openweathermap.org/data/2.5/weather?q=chicago&appid=${apiKey}&units=imperial`;

class Weather extends Component {
    
    constructor(props) 
    {
      super(props);
      this.state = { temp: '',  minTemp: '', maxTemp: ''};
    }

    componentDidMount = ()=> { this.getWeather() };
    
    getWeather() {
        axios.get(url)
        .then(response => { 
            console.log(response);
            this.setState({ 
                temp: response.data.main.temp,
                minTemp: response.data.main.temp_min, 
                maxTemp: response.data.main.temp_max
            });
        })
        .catch(error => {
            console.log(error);
        });
    }

    render() {
      return (
        <div className="Weather">
            <p>Current Temp (Chicago): <span>{this.state.temp}°</span></p>
            <p>High: <span>{this.state.maxTemp}°</span></p>
            <p>Low: <span>{this.state.minTemp}°</span></p>
        </div>
      );
    }
  }
  
  export default Weather;