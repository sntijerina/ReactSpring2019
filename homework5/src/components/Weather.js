import React, { Component } from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux"

import { actionCreators } from '../actions/weatherAction';

class Weather extends Component {
    
    render() {
      return (
        <div className="Weather">
            <button type="button" onClick={()=> this.props.actions.getWeather()}> Get Weather</button>
            <p>Current Temp (Chicago): <span>{this.props.temp}°</span></p>
            <p>High: <span>{this.props.temp_max}°</span></p>
            <p>Low: <span>{this.props.temp_min}°</span></p> 

            <p>Is Error: {this.props.isError?"Yes":"No"}</p>
            <p>Error Message: {this.props.errorMessage}</p>
        </div>
      );
    }
  }

  Weather.propTypes = {
    isLoading: PropTypes.bool,
    temp: PropTypes.string,
    temp_max: PropTypes.string,
    temp_min: PropTypes.string,
    isError: PropTypes.bool,
    errorMessage: PropTypes.string,
    actions: PropTypes.shape({ 
      getWeather: PropTypes.func.isRequired
    })
  };

  const mapStateToProps = state => ({
    isLoading: state.weather.isLoading,
    temp: state.weather.temp,
    temp_max: state.weather.temp_max,
    temp_min: state.weather.temp_min,
    isError: state.weather.isError,
    errorMessage: state.weather.errorMessage
  });

  const mapDispatchToProps = dispatch => {
    return {
      actions: bindActionCreators(
        {
          getWeather : actionCreators.getWeather
        },
        dispatch
      )
    };
  };
  
  export default connect(
    mapStateToProps,
    mapDispatchToProps
  ) (Weather);