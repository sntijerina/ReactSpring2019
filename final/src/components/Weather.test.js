import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Weather from './Weather';
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe('Weather', () => {
  it('renders', () => {
    const component = shallow(<Weather />);
  
    expect(component).toMatchSnapshot();
  });
});