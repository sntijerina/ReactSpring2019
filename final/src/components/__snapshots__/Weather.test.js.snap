// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather renders 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
  spacing={0}
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <WithStyles(Grid)
    item={true}
    xs={3}
  >
    <WeatherForm
      onFormSubmit={[Function]}
    />
    <WeatherCard
      humidity=""
      icon=""
      maxTemp=""
      minTemp=""
      name=""
      overcast=""
      pressure=""
      temp=""
      wind=""
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
